### PR TITLE
support getting various document ranges from InteractiveSessionProviders 

### PR DESCRIPTION
**Pull request by ulugbekna**
In this PR, we add support for getting various documents ranges from `InteractiveSessionProvider`s as part of `provideResponseWithProgress`. This allows passing that information to mapped-edit providers (aka code mappers). 

We also update the API of mappedEditsProvider's.

Support for persisting document ranges across vscode sessions will happen in another PR, if that's okay with @jrieken and @roblourens

### CODE REVIEW

File: src/vscode-dts/vscode.proposed.interactive.d.ts [hunk 1]
```diff
@@ -138,7 +138,22 @@ declare module 'vscode' {
 		treeData: FileTreeData;
 	}
 
-	export type InteractiveProgress = InteractiveProgressContent | InteractiveProgressId | InteractiveProgressTask | InteractiveProgressFileTree;
+	export interface DocumentContext {
```
Comment (roblourens) [inline src/vscode-dts/vscode.proposed.interactive.d.ts:141]: Does this work if there are multiple codeblocks? Does the context need to be associated with one of the codeblocks somehow?

File: src/vs/workbench/contrib/chat/common/chatModel.ts [hunk 2]
```diff
@@ -162,6 +186,8 @@ export class Response implements IResponse {
 		} else if (isCompleteInteractiveProgressTreeData(responsePart)) {
 			this._responseParts.push(responsePart);
 			this._updateRepr(quiet);
+		} else if ('documents' in responsePart) {
+			this._usedContext = responsePart;
```
Comment (roblourens) [inline src/vs/workbench/contrib/chat/common/chatModel.ts:190]: The API would imply that it's valid to send multiple UsedContext pieces

File: src/vscode-dts/vscode.proposed.interactive.d.ts (hunk 1 above)
Comment (ulugbekna) [reply src/vscode-dts/vscode.proposed.interactive.d.ts:141]: We intend for this context to be associated with all codeblocks that occur a single reply for now. We can try associating a code block in the reply with ranges in documents but it's farther in the future.

File: src/vs/workbench/contrib/chat/common/chatModel.ts (hunk 2 above)
Comment (ulugbekna) [reply src/vs/workbench/contrib/chat/common/chatModel.ts:190]: This occurred to me as well. I think it makes sense in future to aggregate all `IUsedContext` progress reports. What do you think?

**Comment by ulugbekna** [comment]
There are some integration tests related to mapped edits failing, I'll soon fix them

**Review by jrieken — APPROVED** [review]

**Pull request merged**